internal/sql/repository/helper/AppListingRepositoryQueryBuilder.go

```diff
@@ -288,9 +288,26 @@ func (impl AppListingRepositoryQueryBuilder) buildAppListingWhereCondition(appLi
 		whereCondition = whereCondition + "and dga.deployment_group_id = " + strconv.Itoa(appListingFilter.DeploymentGroupId) + " "
 	}
 	//add app-status filter here
+	var appStatusExcludingNotDeployed []string
+	var isNotDeployedFilterApplied bool
 	if len(appListingFilter.AppStatuses) > 0 {
-		appStatuses := util.ProcessAppStatuses(appListingFilter.AppStatuses)
-		whereCondition = whereCondition + "and aps.status IN (" + appStatuses + ") "
+		for _, status := range appListingFilter.AppStatuses {
+			if status == "NOT DEPLOYED" {
+				isNotDeployedFilterApplied = true
+			} else {
+				appStatusExcludingNotDeployed = append(appStatusExcludingNotDeployed, status)
+			}
+		}
+	}
+	appStatuses := util.ProcessAppStatuses(appStatusExcludingNotDeployed)
+	if isNotDeployedFilterApplied {
+		whereCondition = whereCondition + " and (p.deployment_app_created=false and p.deployment_app_type != 'manifest_download' or a.id NOT IN (SELECT app_id from pipeline) "
```

Comment:
use fmt.Sprintf and place the variables in where condition with string placeholders

Reply:
also use const for placeholders